cmd/run.go

```diff
@@ -53,15 +58,21 @@ func runRun() error {
 		return err
 	}
 
-	err = rr.Run(context.Background())
+	cancelCtx, cancelFunc := context.WithTimeout(context.Background(), runOpts.streamDuration)
```

Comment:
I have created this new context with the timeout set to stream-duration value which is used across the methods & functions here.